### DISCUSSION TITLE
How do I import in a third-party react component library?

### DISCUSSION THREAD

**Discussion by allen0234** [web-infra-dev/rspress#493] [Q&A] [answered]
如题。如何引入第三方react组件库，如arco design、ant design等？

**Comment by sanyuan0704** [comment]
mdx 里面是可以引入 React 组件的，可以直接用

**Reply by allen0234** [reply]
可以全局按需引入吗？

**Reply by sanyuan0704** [reply]
什么是全局按需引入？可以展示下用法不 @allen0234

**Reply by allen0234** [reply]
写错了，是按需加载。页面里面不用引入，可以直接使用组件。https://arco.design/react/docs/start#%E6%8C%89%E9%9C%80%E5%8A%A0%E8%BD%BD

**Reply by sanyuan0704** [reply, marked as answer]
首先「按需加载」 不等于「页面里面不用引入，可以直接使用组件」。

不过你说的这两个场景 Rspress 都支持，你可以通过 `builderConfig.source.transformImport` 来配置按需加载的插件，配置可以看 https://rsbuild.dev/config/source/transform-import。

如果页面里面不引入直接使用组件，可以使用  Rspress 的 `markdown.globalComponents` 配置，文档：https://rspress.dev/zh/api/config/config-build.html#markdownglobalcomponents
